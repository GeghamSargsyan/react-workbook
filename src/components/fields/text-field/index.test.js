/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import { TextFild } from './index';

const defaultProps = {
  type: 'text',
  theme: 'with-border',
  className: '',
  errorMsg: '',
  name: 'email',
  value: '',
  disabled: false,
  placeholder: 'text',
  onChange: () => {},
  validator: () => '',
};

const BaseTextFild = (props) => <TextFild {...defaultProps} {...props} />;

describe('text fild component', () => {
  it('render correctly', () => {
    const TextFildComponent = renderer.create(<BaseTextFild />).toJSON();
    expect(TextFildComponent).toMatchSnapshot();
  });
  it('render correctly without props', () => {
    const TextFildComponent = renderer.create(<TextFild />).toJSON();
    expect(TextFildComponent).toMatchSnapshot();
  });
  it('check onChange', () => {
    const TextFildComponent = mount(<BaseTextFild />);
    // TextFildComponent.simulate('change');
    TextFildComponent.mount('onchange', { target: { value: 'text' } });
    expect(defaultProps.onChange).toBeTruthy();
  });
});
