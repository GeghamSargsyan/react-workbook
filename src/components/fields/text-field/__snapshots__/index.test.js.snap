// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text fild component render correctly 1`] = `
<div
  className="text-fild"
>
  <input
    className="text-fild_input text-fild_input_with-border"
    disabled={false}
    name="email"
    onChange={[Function]}
    placeholder="text"
    type="text"
    value=""
  />
  <span
    className="text-fild_input_error"
  >
    
  </span>
</div>
`;

exports[`text fild component render correctly without props 1`] = `
<div
  className="text-fild"
>
  <input
    className="text-fild_input text-fild_input_defaulte"
    disabled={false}
    name=""
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
  <span
    className="text-fild_input_error"
  >
    
  </span>
</div>
`;
